sound/soc/sof/ipc4-topology.c

```diff
@@ -1735,8 +1735,11 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
-	memcpy(&process->output_format, &available_fmt->output_pin_fmts[ret].audio_fmt,
-	       sizeof(struct sof_ipc4_audio_format));
+	/* copy Pin 0 output format */
+	if (available_fmt->num_output_formats && ret < available_fmt->num_output_formats &&
+	    !available_fmt->output_pin_fmts[ret].pin_index)
```

Comment:
nitpick: it would be cleaner to `available_fmt->output_pin_fmts[ret].pin_index == 0` as the comment is also telling that we are looking for pin0

Reply:
pin_index is a u32, so !available_fmt->output_pin_fmts[ret].pin_index == 0 is doing the same no?

Reply:
I know, they do the same thing, I was just saying that in this context `!pin_index` is not easily translates to the intention of `pin_index == 0`.
It was a nitpick observation.